TestExpireNSEServer_RefreshFailure is unstable

## Logs
```
--- FAIL: TestExpireNSEServer_RefreshFailure (1.00s)
    nse_server_test.go:182: 
        	Error Trace:	nse_server_test.go:182
        	Error:      	Condition never satisfied
        	Test:       	TestExpireNSEServer_RefreshFailure
```

## Build

https://github.com/networkservicemesh/sdk/pull/739/checks?check_run_id=2001948590

--- a/pkg/registry/common/expire/nse_server.go
+++ b/pkg/registry/common/expire/nse_server.go
@@ -115,10 +115,10 @@ func (n *expireNSEServer) newTimer(
 
 	t.timer = time.AfterFunc(time.Until(expirationTime), func() {
 		t.executor.AsyncExec(func() {
+			t.started = true
 			if t.canceled || n.ctx.Err() != nil {
 				return
 			}
-			t.started = true
 
 			unregisterCtx, cancel := context.WithCancel(n.ctx)
 			defer cancel()

